dubbo-registry/dubbo-registry-api/src/main/java/org/apache/dubbo/registry/support/AbstractRegistry.java

```diff
@@ -356,7 +352,7 @@ protected void notify(List<URL> urls) {
             }
 
             Set<NotifyListener> listeners = entry.getValue();
-            if (listeners != null) {
+            if (listeners != null && !listeners.isEmpty()) {

```

Comment:
I think !listeners.isEmpty() will be extra, below for loop will be skiped if empty. So we can relax on this extra check. What do you say?